File: abac/tests/example_scripts/python/abac_keyid.py
#!/usr/bin/env python

"""
  abac_keyid.py

   To demonstrate how to use ABAC's api in python to access keyid of a 
principal credential

   pre-condition: generate IceCream_ID.pem and IceCream_private.pem with
           creddy --generate --cn IceCream

   keyid of the loaded principal credential is printed

"""

from sys import argv, exit
from ABAC import ID

if len(argv) < 2:
    print "Usage: abac_keyid.py <cert.pem>"
    exit(1)

id = None
try:
    print argv[1]
    id = ID(argv[1])
except Exception, e:
    print "Problem loading cert: %s" % e
    exit(1)

print id.keyid()
print "okay"

File: abac/tests/example_scripts/perl/abac_keyid.pl
#!/usr/bin/env perl

use ABAC;

my $filename = shift || die "Usage: abac_keyid.pl <cert.pem>\n";

my $id;
eval {
    # will throw a RuntimeException if it can't load the cert
    $id = ABAC::ID->new($filename);
};
if ($@) {
    print "ERROR, Problem loading cert: $@";
    exit;
}

print $id->keyid, "    OKAY\n";
